shared/src/Datadog.Trace.ClrProfiler.Native/crashhandler.cpp

```diff
@@ -0,0 +1,330 @@
+        {
+            Log::Warn("Crashtracking - Trying to initialize the crash handler twice");
+            return false;
+        }
+
+        auto dllPath = GetCurrentDllPath();
+
+        if (dllPath.empty())
+        {
+            Log::Warn("Crashtracking - Could not register the crash handler: error when retrieving the path of the DLL");
+            return false;
+        }
+
+        GetDatadogEnvironmentBlock(_context.Environ, _context.EnvironLength);
+
+        // Register the crash handler in the registry
+        // Windows expects a DWORD value with the full path of the DLL as the name, 
+        // in HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\Windows Error Reporting\RuntimeExceptionHelperModules
+        HKEY hKey;
+        LPCWSTR subKey = L"SOFTWARE\\Microsoft\\Windows\\Windows Error Reporting\\RuntimeExceptionHelperModules";
+        DWORD value = 1;
+
+        // Open the key
+        DWORD disposition;
+        auto result = RegCreateKeyEx(HKEY_CURRENT_USER, subKey, 0, NULL, 0, KEY_SET_VALUE, NULL, &hKey, &disposition);
+
+        if (result != ERROR_SUCCESS)
+        {
+            return false;
+        }
+
+        // Set the value
+        RegSetValueEx(hKey, dllPath.c_str(), 0, REG_DWORD, reinterpret_cast<const BYTE*>(&value), sizeof(value));
+        RegCloseKey(hKey);
+
+        // The profiler API is not initialized yet, so we look for clr.dll/coreclr.dll
+        // to know if we're running with .NET Framework or .NET Core
+        bool isDotnetCore = true;
+        HMODULE hModule = GetModuleHandle(L"coreclr.dll");
+
+        if (hModule == NULL)
+        {
+            hModule = GetModuleHandle(L"clr.dll");
+            isDotnetCore = false;
+
+            if (hModule == NULL)
+            {
+                Log::Warn("Crashtracking - Failed to get module handle for coreclr.dll or clr.dll");
+                return false;
+            }
+        }
+
+        wchar_t buffer[MAX_PATH];
+
+        if (GetModuleFileNameW(hModule, buffer, MAX_PATH) == 0)
+        {
+            Log::Warn("Crashtracking - Failed to get module filename");
+            return false;
+        }
+
+        auto clrFileName = std::wstring(buffer);
+
+        std::filesystem::path clrFileNamePath(clrFileName);
+        auto clrDirectory = clrFileNamePath.parent_path();
+
+        // Build the path to the DAC (mscordacwks.dll on .NET, mscordaccore.dll on .NET Core)
+        std::wstring dacFileName = isDotnetCore ? L"mscordaccore.dll" : L"mscordacwks.dll";
+        std::filesystem::path dacFilePath = clrDirectory / dacFileName;
+
+        // Unregister the .NET handler
+        Log::Debug("Crashtracking - Unregistering the .NET handler ", dacFilePath.c_str());
+        auto unregisterDacHr = WerUnregisterRuntimeExceptionModule(dacFilePath.c_str(), (PVOID)hModule);
+
+        if (FAILED(unregisterDacHr))
+        {
+            Log::Warn("Crashtracking - Failed to unregister the DAC handler: ", unregisterDacHr);
```

Comment:
is it safe that we register our handler AFTER the CLR one?

Reply:
It just means that we probably won't be called. But we should still register ours just in case (the .NET handler might not claim the crash, or maybe it wasn't registered to begin with for an unknown reason)